axelrod/tests/unit/test_fingerprint.py

```diff
@@ -0,0 +1,133 @@
+                    (0.0, 0.5),
+                    (0.0, 0.75),
+                    (0.25, 0.0),
+                    (0.25, 0.25),
+                    (0.25, 0.5),
+                    (0.25, 0.75),
+                    (0.5, 0.0),
+                    (0.5, 0.25),
+                    (0.5, 0.5),
+                    (0.5, 0.75),
+                    (0.75, 0.0),
+                    (0.75, 0.25),
+                    (0.75, 0.5),
+                    (0.75, 0.75)]
+long_probes = [JossAnnTransformer(c)(probe)() for c in long_coordinates]
+long_edges = [(0, 2),
+              (0, 3),
+              (0, 4),
+              (0, 5),
+              (0, 6),
+              (0, 7),
+              (0, 8),
+              (0, 9),
+              (0, 10),
+              (0, 11),
+              (0, 12),
+              (1, 13),
+              (0, 14),
+              (0, 15),
+              (1, 16),
+              (1, 17)]
+
+
+class TestFingerprint(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.strategy = strategy
+        cls.probe = probe
+        cls.expected_probes = probes
+        cls.expected_coordinates = coordinates
+        cls.expected_edges = edges
+
+    def test_init(self):
+        strategy = axl.Cooperator()
+        probe = axl.TitForTat()
+        fingerprint = AshlockFingerprint(strategy, probe)
+        self.assertEqual(fingerprint.strategy, strategy)
+        self.assertEqual(fingerprint.probe, probe)
+
+    def test_create_coordinates(self):
+        test_coordinates = create_coordinates(0.5)
+        coordinates = [(0.0, 0.0), (0.0, 0.5), (0.5, 0.0), (0.5, 0.5)]
+        self.assertEqual(test_coordinates, coordinates)
+
+    def test_create_probes(self):
+        coords = [(0.0, 0.0), (0.0, 0.5), (0.5, 0.0), (0.5, 0.5)]
+        probes = create_probes(probe, coords)
+        self.assertEqual(len(probes), 4)
+        # self.assertEqual(probes, self.expected_probes)
+
+    def test_create_edges(self):
+        coords = [(0.0, 0.0), (0.0, 0.5), (0.5, 0.0), (0.5, 0.5)]
+        expected_edges = [(0, 2), (0, 3), (0, 4), (0, 5)]
+        edges = create_edges(coords)
+        self.assertEqual(edges, expected_edges)
+
+    def test_construct_tournament_elemets(self):
+        af = AshlockFingerprint(self.strategy, self.probe)
+        edges, tournament_players = af.construct_tournament_elements(0.5)
+        self.assertEqual(edges, self.expected_edges)
+        self.assertEqual(len(tournament_players), 6)
+        self.assertEqual(tournament_players[0].__class__, af.strategy)
+
+    def test_fingerprint(self):
+        af = AshlockFingerprint(self.strategy, self.probe)
+        af.fingerprint(turns=10, repetitions=2, step=0.25, processes=1)
+        edge_keys = sorted(list(af.interactions.keys()))
+        coord_keys = sorted(list(af.data.keys()))
+        self.assertEqual(af.step, 0.25)
+        self.assertEqual(edge_keys, sorted(long_edges))
+        self.assertEqual(coord_keys, long_coordinates)
```

Comment:
Can we also test the generated data? (Basically anything that fingerprint creates, can we check it).

Could you also do a check that fingerprint works on all strategies (obviously for a huge size, tiny turns etc...). I suggest you do this in two ways:

- Write a hypothesis test (I can help) that picks two strategies, sets one to be the probe and checks that the data is of the correct form (this will also check that it ran). 
- Write a loop that just checks them all. I think that this would be my preferred option.

I think this is important to have, in the same way that we have tests on tournaments that the full tournament runs for example.